plugins/grafana-custom-plugins/grafana-http-table-plugin/src/components/TablePanel.tsx

```diff
@@ -0,0 +1,136 @@
+  const destinationTransportPorts = frame.fields.find((field) => field.name === 'destinationTransportPort');
+  const httpValsList = frame.fields.find((field) => field.name === 'httpVals');
+  const defaultMaterialTheme = createTheme({palette: {
+    mode: config.theme2.isDark ? 'dark': 'light',
+  }});
+
+  let tableData: FlowRow[] = [];
+  interface FlowRow {
+    source: string,
+    destination: string,
+    txId: number,
+    hostname: string,
+    url: string,
+    http_user_agent: string,
+    http_content_type: string,
+    http_method: string,
+    protocol: string,
+    status: number,
+    length: number,
+    subRows: FlowRow[],
+  }
+  for (let i = 0; i < frame.length; i++) {
+    const sourceIP = sourceIPs?.values.get(i);
+    const destinationIP = destinationIPs?.values.get(i);
+    const sourcePort = sourceTransportPorts?.values.get(i);
+    const destinationPort = destinationTransportPorts?.values.get(i);
+    const httpVals = httpValsList?.values.get(i);
+    let httpValsJSON: any;
+    if (httpVals !== undefined) {
+      httpValsJSON = JSON.parse(httpVals);
+    }
+    const rowData: FlowRow = {
+      source: '',
+      destination: '',
+      txId: 0,
+      hostname: '',
+      url: '',
+      http_user_agent: '',
+      http_content_type: '',
+      http_method: '',
+      protocol: '',
+      status: 0,
+      length: 0,
+      subRows: [],
+    };
+    function setTableRow(source: string, destination: string, txId: number, hostname: string, url: string, http_user_agent: string, http_content_type: string, http_method: string, protocol: string, status: number, length: number) {
+      if (txId === 0) {
+        rowData.source = source;
+        rowData.destination = destination;
+        rowData.txId = txId;
+        rowData.hostname = hostname;
+        rowData.url = url;
+        rowData.http_user_agent = http_user_agent;
+        rowData.http_content_type = http_content_type;
+        rowData.http_method = http_method;
+        rowData.protocol = protocol;
+        rowData.status = status;
+        rowData.length = length;
+      } else {
+        const row: FlowRow = {
+          source: source,
+          destination: destination,
+          txId: txId,
+          hostname: hostname,
+          url: url,
+          http_user_agent: http_user_agent,
+          http_content_type: http_content_type,
+          http_method: http_method,
+          protocol: protocol,
+          status: status,
+          length: length,
+          subRows: [],
+        }
+        rowData.subRows.push(row);
+      }
+    }
+    for (const txId in httpValsJSON) {
+      setTableRow(sourceIP+':'+sourcePort, destinationIP+':'+destinationPort, +txId, httpValsJSON[txId].hostname, httpValsJSON[txId].url, httpValsJSON[txId].http_user_agent, httpValsJSON[txId].http_content_type, httpValsJSON[txId].http_method, httpValsJSON[txId].protocol, httpValsJSON[txId].status, httpValsJSON[txId].length);
```

Comment:
Why `+txId` instead of `txId`? Isn't `txId` already being a number here?

Reply:
since i'm interpreting the JSON, compiler thinks txID could have type string, and in order to convert it, I use the '+'